File: src/main/java/org/hwabeag/hwaskyblock/schedules/UnloadBorderTask.java
package org.hwabeag.hwaskyblock.schedules;

import org.bukkit.Bukkit;
import org.bukkit.World;
import org.bukkit.WorldBorder;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.hwabeag.hwaskyblock.config.ConfigManager;

import java.util.Objects;

public class UnloadBorderTask implements Runnable {

    FileConfiguration SkyBlockConfig = ConfigManager.getConfig("skyblock");

    @Override
    public void run() {
        for (Player player : Bukkit.getOnlinePlayers()) {
            World world = player.getWorld();
            String world_name = world.getWorldFolder().getName();
            String[] number = world_name.split("\\.");
            if (Objects.equals(number[0], "HwaSkyBlock")) {
                String id = number[1];
                setWorldBorder(world.getName(), SkyBlockConfig.getInt(id + ".size"));
            }
        }
    }

    public void setWorldBorder(String worldName, int size) {
        World world = Bukkit.getWorld(worldName);
        if (world != null) {
            WorldBorder border = world.getWorldBorder();
            border.setCenter(world.getSpawnLocation());
            border.setSize(size);
            border.setWarningDistance(10);
            border.setDamageAmount(0.1);
        }
    }
}

File: src/main/kotlin/schedules/UnloadBorderTask.kt
package org.hwabeag.hwaskyblock.schedules

import org.bukkit.Bukkit
import org.bukkit.configuration.file.FileConfiguration
import org.hwabeag.hwaskyblock.database.DatabaseManager
import org.hwabeag.hwaskyblock.database.config.ConfigManager

class UnloadBorderTask : Runnable {

    override fun run() {
        for (player in Bukkit.getOnlinePlayers()) {
            val world = player.world
            val world_name = world.worldFolder.getName()
            val number: Array<String?> = world_name.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            if (number[0] == "HwaSkyBlock") {
                val id = number[1]
                val size = DatabaseManager
                    .getSkyBlockData(id.toString(), "$id.size", "getSkyBlockSize")
                    ?.toString()?.toIntOrNull() ?: 0
                setWorldBorder(world.name, size)
            }
        }
    }

    fun setWorldBorder(worldName: String?, size: Int) {
        val world = worldName?.let { Bukkit.getWorld(it) }
        if (world != null) {
            val border = world.worldBorder
            border.center = world.spawnLocation
            border.size = size.toDouble()
            border.warningDistance = 10
            border.damageAmount = 0.1
        }
    }
}